app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.search
+
+import android.content.Context
+import androidx.navigation.NavController
+import mozilla.components.browser.search.SearchEngine
+import mozilla.components.browser.search.SearchEngineManager
+import mozilla.components.support.ktx.kotlin.isUrl
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.search.awesomebar.AwesomeBarInteractor
+import org.mozilla.fenix.search.toolbar.ToolbarInteractor
+
+/**
+ * Helper function to get the MetricController off of context.
+ */
+val Context.metrics: MetricController
```

Comment:
Could we move these utility extension functions out to org.mozilla.fenix.ext.Context.kt? They aren't easy to find in a file specific to the search feature.